sla_per_user_system_test.py

```diff
@@ -0,0 +1,501 @@
+            user = entity.get('user')
+            if user:
+                user.drop()
+            if role:
+                role.drop()
+            if sl:
+                sl.drop()
+
+        self.backgroud_task = None
+        self.connection_cql.cluster.shutdown()
+
+    def fill_cache_before_test(self, max_key_for_read, stress_duration):
+        read_cmds = [self.STRESS_READ_CMD.format(n=self.n, user=self.DEFAULT_USER,
+                                                 password=self.DEFAULT_USER,
+                                                 pop="seq=1..%d" % max_key_for_read,
+                                                 duration='%dm' % stress_duration,
+                                                 threads=200)
+                     ]
+        self.run_stress_and_verify_threads(params={'stress_cmd': read_cmds})
+
+    def define_read_cassandra_stress_command(self, user, load_type, workload_type, threads, stress_duration_min,
+                                             max_rows_for_cache_read=None):
+        """
+        :param user: dict with User/Role/ServiceLevel objects
+        :param load_type: cache_only/disk_only/mixed
+        :param workload_type: latency: with ops restriction - using throttle
+                                or
+                              throughput: no restriction
+        """
+        def latency():
+            return '%d throttle=20000/s' % threads
+
+        def throughput():
+            return threads
+
+        def cache_only(max_rows_for_cache_read):
+            if not max_rows_for_cache_read:
+                max_rows_for_cache_read = self.n * 0.25
+            return 'seq=1..%d' % max_rows_for_cache_read
+
+        def mixed(max_rows_for_cache_read):
+            if not max_rows_for_cache_read:
+                max_rows_for_cache_read = self.n
+            return "'dist=gauss(1..%d, %d, %d)'" % (max_rows_for_cache_read,
+                                                    max_rows_for_cache_read / 2,
+                                                    max_rows_for_cache_read * 0.05)
+
+        def disk_only(max_rows_for_cache_read):
+            if not max_rows_for_cache_read:
+                max_rows_for_cache_read = self.n * 0.3
+            return 'seq=%d..%d' % (max_rows_for_cache_read+(self.n*0.3), self.n)
+
+        user_name = user['user'].name
+
+        rate = locals()[workload_type]()  # define -rate for c-s command depend on workload type
+        pop = locals()[load_type](max_rows_for_cache_read)  # define -pop for c-s command depend on load type
+
+        c_s_cmd = self.STRESS_READ_CMD.format(n=self.n, user=user_name,
+                                              password=user_name,
+                                              pop=pop,
+                                              duration='%dm' % stress_duration_min,
+                                              threads=rate)
+        return c_s_cmd
+
+    def test_two_slas_one_to_five_ratio_mix_load(self):
```

Comment:
if you mean by "mix" to (cache_and_disk) please change it, in our terms mix is a mix between read and write.
I see here that both users are doing read.

Reply:
changed